### ISSUE TITLE
lower performance with cc miner

### ISSUE THREAD

**Issue by gatistaabb** [MultiPoolMiner/MultiPoolMiner#821]
If I start miner from a bat file it makes 27.76MH/s and with multipool miner it makes 24.68MH/s.

Same miner, same starting command.

ccminer TPruvot - algo x17

Why?

**Comment by gatistaabb** [comment]
More details: 3 1070's

line for ccminer bat file: 
ccminer-x64.exe -a x17 -o stratum+tcp://x17.mine.zpool.ca:3737 -u 1CGRGAj6wJ8w9DnSLd5pzTeZYJVnwQRQa3 -p v1070m1,c=BTC -t 3

from miners - no custom setup:

$Commands | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty Name | Where-Object {$Pools.(Get-Algorithm $_).Protocol -eq "stratum+tcp" <#temp fix#>} | ForEach-Object {
    [PSCustomObject]@{
        Type = "NVIDIA"
        Path = $Path
        Arguments = "-a $_ -o $($Pools.(Get-Algorithm $_).Protocol)://$($Pools.(Get-Algorithm $_).Host):$($Pools.(Get-Algorithm $_).Port) -u $($Pools.(Get-Algorithm $_).User) -p $($Pools.(Get-Algorithm $_).Pass)$($Commands.$_) -t 3"
        HashRates = [PSCustomObject]@{(Get-Algorithm $_) = $Stats."$($Name)_$(Get-Algorithm $_)_HashRate".Week}
        API = "Ccminer"
        Port = 4068
        Wrap = $false
        URI = $Uri
    }
}

**Comment by gatistaabb** [comment]
How do I know that? I have one computer with 1 1070 and one rig with 3 1070 same models.

On the computer im making 9.07MH/s and on the rig im making only 24.68MH/s.

When I start from bat file I get the expected 27.76MH/s. When started from multiminer it mines 24.68 only.

Why?

**Comment by gatistaabb** [comment]
found some answers at #522

**Comment by gatistaabb** [comment]
What intensity is recommended for 3 1070?

**Comment by gatistaabb** [comment]
Still cant find a value that works! Need some help here

**Comment by UselessGuru** [comment]
This does not make any sense. MPM just launches miners based on best profitability.
The only reason for this to happen could by CPU mining the same time. This could slow down your entire system. -> Do not use CPU mining (or at least run the CPU miner at very low priority).
BTW: Forget all the X.. algorithms. They are not profitable.

**Comment by gatistaabb** [comment]
X17 is the chosen algo by MPM tô mine on zpool. I Just compared the hash rate

**Comment by gak45** [comment]
Yeah X17 and Sib (x111 ghost) and Xevan are pretty profitable alogs right now. Could you (uselessGuru) give a bit more details on why they should be yanked from 1070 an 1080 mining rigs if they appear to be profitable?
To fix this profit issue I simply deleted the X17 hashrate file in the 'stats' folder and it rebenchmarked it (a lower number now) and is working fine. Might solve your issue for you too :)

**Comment by gatistaabb** [comment]
@gak45 already did it like 100 times. Trust me.

Cant find any answers! What i'm trying is to mine on bat file x11 and let the MPM mine on amd

**Comment by aaronsace** [comment]
MultiPoolMiner doesn't effect/control the speed. MultiPoolMiner just launchers the miners and has no control over how fast they run. 

You have most likely got too many GPUs and not enough CPU power.

**Comment by gatistaabb** [comment]
@aaronsace I have 3 1070's and 1 r9 380x

My cpu is i3-7100. Is that not enought?

**Comment by aaronsace** [comment]
It's only a dual core, right? 

What else could the problem possibly be?

**Comment by gatistaabb** [comment]
If the problem is the CPU, MPM is making it happens.

Cause when I open the ccminer from a bat file, it works well.

**Comment by aaronsace** [comment]
What can we do if your CPU is maxing out...?

**Comment by gatistaabb** [comment]
But when I open task manager it shows not even 30% (with team viewer open that eats 10% CPU). It still can be the CPU?

**Comment by aaronsace** [comment]
@gatistaabb Beyond me, I've got nothing.

**Comment by fonyo** [comment]
@gak45 No, it should not be in the README, the scope of what you are asking for is huge and so far just optimistically planned for V3, it also requires an insane amount of research in terms of detailed config recorded and applied safely per GPU. There have only been discussions around this and everything else we've got is highly experimental.

What is not clear to me and have not been asked so far:
Are you using the tpruvot miner located in the /Bin folder or something that you downloaded separately? Please test using the same executable file/config that MPM uses.

**Comment by gatistaabb** [comment]
@fonyo using the tpruvot miner on bin folder.

And the same executable line that appears on MPM under running

**Comment by fonyo** [comment]
Post a log please, want to see the start.bat

**Comment by fonyo** [comment]
And the related stat files as well please

**Comment by gatistaabb** [comment]
Follows the logs:
[Log.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1586213/Log.txt) -> log from the MPM
[CcminerTpruvot.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1586214/CcminerTpruvot.txt) 
-> ccminerTpruvot.ps1
[LogCCminer.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1586215/LogCCminer.txt) -> Log from the bat cc miner located on bin folder
[CcminerTpruvot_X17_HashRate.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1586216/CcminerTpruvot_X17_HashRate.txt) -> X17 HashRate - keep in mind I deleted this many times trying to get right values, but it still shows less than the expected

**Comment by fonyo** [comment]
It takes over 30 minutes for the standalone tpruvot to ramp up to 27. You only had 13 minutes of tpruvot mining (combined, with breaks) with MPM. MPM switches away before it could ramp up due to coin spiking all the time. Leave it for a few days without clearing anything and you may probably see MPM completely ignoring these algos due to their instability.

**Comment by fonyo** [comment]
Even though zpool operates with their special pplns++ payout method, switching is still bad in a way but necessary! MPM will try to filter these unstable coins for you so you are mining the best possible with the least switching and with more continuous mining.

**Comment by gatistaabb** [comment]
Well, I before I reset everything I left it for a few days. Then I noticed the difference.

Another thing I notice is on the ccMiner window that MPM opens, there are lots of drops on the GPU hash, much more than the stand alone.

If you check the standalone, there are few drops on GPU's hashrate below 9000 kH/s, while on the MPM ccMiner window I could see lots of 7000 kH/s every now and then. I tried to log it but couldnt.

I openned a Issue about logging ccMiner of MPM and it got closed without a right answer. If you could check it out and maybe reopen the issue i'd appreciate. #820

**Comment by fonyo** [comment]
So far my comment above still explains the reason for what you experience.

**Comment by gatistaabb** [comment]
Ok, I'll reset everything again and will let it roll for a couple of days and resend log if the problem persists.

**Comment by fonyo** [comment]
Just checked zpool for current/predicted and whattomine for past profitability data and the algos mentioned above are not in the top 5. Please don't forget what I stated in the other comment. Mine the most profitable. That's the whole point of MPM.

**Comment by gatistaabb** [comment]
I noticed that too, but I tought since MPM takes my hashrate into account it was mining the most profitable coin for my hardware. For a few days it keeps switching between blake2s, x17. Have I done something wrong?

**Comment by gak45** [comment]
@fonyo I used only the ones MPM installed. It just goes squirrely sometimes and a reset and re-bench fixes it. 

I just wanted to clarification on the recommendation you threw out there on X alogs not being profitable for NV rigs. Maybe I am missing something. You are the expert so just wanted to know your reasoning and POV on said recommendation that you publicly made. 

You are asking (well actually demanding) donation mining and yet don't seem to be in a hurry to clarify any recommendations... which seem counter intuitive as it is in your own best interest to have those 24minutes of mining donation be spent on profitable alogs!

**Comment by fonyo** [comment]
@gak45 Sorry I meant address @gatistaabb with my original post. My bad.

However, please allow me to clarify a few things regarding everything that is mentioned here.

I cannot put a recommended list of algos in the README as the worst performing algo on any card can become the most profitable one at any time. Things are changing rapidly in this regard. There are the example start.bat files for this reason. They provide you a balanced/recommended list of algorithms for all rigs. The README advises against including too many or all available algorithms. The website will be the forum for such recommendations at some point.

I have been helping users here for months just for fun before I was included in the donation script. I don't have any urge for filling the documentation with financial advice and other useless information just to earn more. At the end of the day, it is upto the users to decide what they want to mine and we need to allow them to do so even if it means they are generating losses compared to sticking with the default settings. The other reason I would not include such information is because that is what the script does. It is its job to exclude such algorithms that are not profitable by not mining them. As stated in the README as well, the script becomes "smarter" as it accumulates data and will ignore coin spikes (this is a very important new feature). This takes time (a few days or a week of mining). Resetting the stats will clear everything it learnt.

I could carry on forever explaining everything MPM does but there is no point doing so. We can provide plenty of advice here and are happy to do so. And again, MPM is meant to find and mine the most profitable algorithm and it takes into account a lot of variables for this (with a lot more to come). It is not for generic mining whatsoever but for pure profit-making.

I hope this clears a few concerns and clarify things. Please try to stick to the default batch files if unsure about how payout systems, coin difficulty/price calculations and the crypto world work as a whole in general with regards to MPM functionality.

**Comment by gatistaabb** [comment]
After some thinking about when you said how much time the x17, for example, needed to built up its hash rate and verifying checking the logs again.

You can see that even on standalone, it takes ~20 minutes to reach 27MH/s, but it was actually mining 27MH/s since the beginning, you can see all 3 GPU's each ~9MH/s.

That build up maybe the screw we're hammering here. It actually is faulty for MPM since it doesnt report the actual hash rate. So MPM sees 24MH/s when actually is achieving 27MH/s because that build up time, as you said.

It makes MPM understands that I can do 24MH/s and maybe mine something that is not that prof compared to the 27MH/s. That is not MPM's fault, I can see that much, but one temporary work around is to have a paramether for each algo for min run time. ie 30 minutes for x17, so it wont display wrong hashes again as "-mt 30" somewhere, some ifs around here and there so it wont change algos (not that hard I'd guess).

**Comment by UselessGuru** [comment]
@gak45 
> I just wanted to clarification on the recommendation that was stated that 'X alogs not being profitable for NV rigs'.

Most X algos are not profitable. How to find out: 
Believe MPM (it does exactly what I describe below)
Or the hard way: read the hashrate of your miner for algo X. Then check the projected revenue for that algo on the pool page. multiply this number with your hashrate -> profit per 24h in BTC

**Comment by UselessGuru** [comment]
@gatistaabb 

> Default is 60." but the benchmarks run for 2 minutes.

MPM will wait 60 seconds and the begin reading the hashrate from the miner.
This will take some time (defined in the API files):
```
$Delta = 0.05
$Interval = 5
$HashRates = @()
...
do {
...
read hash rate here
...
} while ($HashRates.Count -lt 6)
```
inside the loop the API file attempts to read the hashrate from the miner (this sometimes fails - no worries) and then sleep $interval seconds.
It will end the loop after 6 succesful readouts -> benchmarking will take at least 6x5 = 30 seconds -> total loop > 90 seconds

**Comment by gatistaabb** [comment]
I tested what I wrote before and I also openned up MPM code so I can see the Interval like @UselessGuru said. I did -interval 1200 for 20 minutes, so it will check the price on the pool(update pool) once every 20 minutes so it can change algos. So, its enought time to build up everything it needs to.

The algo that was having problem benchmarked to 27.5MH/s. With no problems.

I would suggest min algo time for some algos, some paramether for more control.

Thanks alot for the support!
